### ISSUE TITLE
Leancloud visitors statistic is empty. 阅读人数统计不显示

### ISSUE THREAD

**Issue by songjinze** [theme-next/hexo-theme-next#1668]
Please follow this Issue template to provide relevant information, such as source code repositories, blog links, and screenshots, which will help us investigate.
请按照此 Issue 模版提供相关信息，例如源码仓库、博客链接和屏幕截图，这将有助于我们进行调查。

## Issue Checklist 

- [x] I am using [the latest](https://github.com/theme-next/hexo-theme-next/releases/latest) version of NexT.
- [x] I have read the relevant documents of [Hexo](https://hexo.io/docs/) and [NexT](https://theme-next.org/docs/).
- [x] I have reviewed the latest Roadmap on GitHub and searched for current [issues](https://github.com/theme-next/hexo-theme-next/issues), which does not help me.

***

## Actual behavior 

- Links to demo site with this issue: N/A
- Links to repository or source code of the blog: N/A

## Steps to reproduce the behavior 
1. N/A
2. N/A
3. N/A

### Node.js and NPM Information

```
v13.10.1
7.11.1
```

### Package dependencies Information

```
{
  "name": "hexo-site",
  "version": "0.0.0",
  "private": true,
  "hexo": {
    "version": "3.9.0"
  },
  "dependencies": {
    "hexo": "^3.9.0",
    "hexo-deployer-git": "^0.3.1",
    "hexo-generator-archive": "^0.1.4",
    "hexo-generator-baidu-sitemap": "^0.1.9",
    "hexo-generator-category": "^0.1.3",
    "hexo-generator-feed": "^1.2.2",
    "hexo-generator-index": "^0.2.1",
    "hexo-generator-search": "^2.4.0",
    "hexo-generator-search-zip": "0.0.1",
    "hexo-generator-sitemap": "^1.2.0",
    "hexo-generator-tag": "^0.2.0",
    "hexo-leancloud-counter-security": "^1.5.0",
    "hexo-renderer-ejs": "^0.3.0",
    "hexo-renderer-marked": "^0.3.0",
    "hexo-renderer-njucks": "^1.0.0",
    "hexo-renderer-sass": "^0.3.2",
    "hexo-renderer-stylus": "^0.3.3",
    "hexo-server": "^0.2.0",
    "hexo-symbols-count-time": "^0.7.1",
    "plugin": "^0.3.3"
  }
}

```

### Hexo Configuration

```yml
leancloud_counter_security:
  enable_sync: true
  app_id: <correct>
  app_key: <correct>
  # server_url: 
  # hexo lc-counter register <username> <password>
  username: 
  password: 
```

### NexT Configuration

```yml
# Show number of visitors of each article.
# You can visit https://leancloud.cn to get AppID and AppKey.
# AppID and AppKey are recommended to be the same as valine's for counter compatibility.
# Do not enable both `valine.visitor` and `leancloud_visitors`.
leancloud_visitors:
  enable: true
  app_id:# <my correct app_id>
  app_key:  # <my app_key>
  # Required for apps from CN region
  server_url:  # <my_correct_url>
  # Dependencies: https://github.com/theme-next/hexo-leancloud-counter-security
  # If you don't care about security in leancloud counter and just want to use it directly
  # (without hexo-leancloud-counter-security plugin), set `security` to `false`.
  security: true

# Valine
# For more information: https://valine.js.org, https://github.com/xCss/Valine
valine:
  enable: false
  appid: # Your leancloud application appid
  appkey: # Your leancloud application appkey
  notify: false # Mail notifier
  verify: false # Verification code
  placeholder: Just go go # Comment box placeholder
  avatar: mm # Gravatar style
  guest_info: nick,mail,link # Custom comment header
  pageSize: 10 # Pagination size
  language: # Language, available values: en, zh-cn
  visitor: false # Article reading statistic
  comment_count: false # If false, comment count will only be displayed in post page, not in home page
  recordIP: false # Whether to record the commenter IP
  serverURLs: # When the custom domain name is enabled, fill it in here (it will be detected automatically by default, no need to fill in)
  #post_meta_order: 0
```

## Other Information 

article page got no error but empty.文章页面js consle没有报错，但是view显示是空的。

![image](https://user-images.githubusercontent.com/26564718/132497185-f6b5f806-0097-45f4-888f-a928bd0f06a4.png)

![image](https://user-images.githubusercontent.com/26564718/132497111-a90a7afe-18ae-42e2-8a11-2dfe1466cf7f.png)

![image](https://user-images.githubusercontent.com/26564718/132496877-5cb427a7-283f-44b3-9c1c-d6aaf88ad278.png)

index page got an error and the tag is also empty. 主页面文章简介中，显示的view也是空的，并且console有报错。

![image](https://user-images.githubusercontent.com/26564718/132496927-495ea751-8340-4602-aee6-065b249b8bbe.png)

**Comment by songjinze** [comment]
by the way, I’m using hexo on my blog on github.io page

**Comment by songjinze** [comment]
The reason why article page had no view count was found. It's because i write hexo 's _config.yml url element to www.xxx.github.io instead of https://xxx.github.io. 
But when i changed this, the article page also has the same error where index page got.
![image](https://user-images.githubusercontent.com/26564718/132608864-99fc1f1d-32a7-4a90-be4f-f7b609cda841.png)
![image](https://user-images.githubusercontent.com/26564718/132608904-221829af-42b5-4480-b2cd-deb13388a8c3.png)
Looks like the function in layout/_third-party/statistic/lean-analytics.swig  leancloudSelector(url) doesn't work correctly.
![image](https://user-images.githubusercontent.com/26564718/132609113-4220a1e0-fbdd-48f5-9fae-633c838cca9e.png)

But it's strange why the tag is null? looks like the tag is created by hexo itself..

<blockquote><div><strong><a href="https://xxx.github.io/">woot!</a></strong></div></blockquote>

**Comment by songjinze** [comment]
I have solved this problem by update hexo to 4.2.1 and my package.json is :
```
{
  "name": "hexo-site",
  "version": "0.0.0",
  "private": true,
  "hexo": {
    "version": "4.2.1"
  },
  "dependencies": {
    "hexo": "^4.0.0",
    "hexo-deployer-git": "^0.3.1",
    "hexo-generator-archive": "^0.1.4",
    "hexo-generator-baidu-sitemap": "^0.1.9",
    "hexo-generator-category": "^0.1.3",
    "hexo-generator-feed": "^1.2.2",
    "hexo-generator-index": "^0.2.1",
    "hexo-generator-search": "^2.4.0",
    "hexo-generator-search-zip": "0.0.1",
    "hexo-generator-sitemap": "^2.1.0",
    "hexo-generator-tag": "^0.2.0",
    "hexo-leancloud-counter-security": "^1.5.0",
    "hexo-renderer-ejs": "^0.3.0",
    "hexo-renderer-marked": "^0.3.0",
    "hexo-renderer-njucks": "^1.0.0",
    "hexo-renderer-sass": "^0.3.2",
    "hexo-renderer-stylus": "^0.3.3",
    "hexo-server": "^0.2.0",
    "hexo-symbols-count-time": "^0.7.1",
    "plugin": "^0.3.3"
  }
}

```

looks like the tage url using chinese is wrong with something like %20%E4. And the new version of hexo fixed that problem!
